src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py

```diff
@@ -0,0 +1,161 @@
+"""Zowe Python Client SDK.
+
+This program and the accompanying materials are made available under the terms of the
+Eclipse Public License v2.0 which accompanies this distribution, and is available at
+
+https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zowe Project.
+"""
+
+import os
+
+from zowe.core_for_zowe_sdk import SdkApi
+from zowe.core_for_zowe_sdk.exceptions import FileNotFound
+
+_ZOWE_FILES_DEFAULT_ENCODING = "utf-8"
+
+class USSFiles(SdkApi):
+    """
+    Class used to represent the base z/OSMF USSFiles API
+    which includes all operations related to USS files.
+
+    ...
+
+    Attributes
+    ----------
+    connection
+        connection object
+    """
+
+    def __init__(self, connection):
+        """
+        Construct a USSFiles object.
+
+        Parameters
+        ----------
+        connection
+            The z/OSMF connection object (generated by the ZoweSDK object)
+
+        Also update header to accept gzip encoded responses
+        """
+        super().__init__(connection, "/zosmf/restfiles/", logger_name=__name__)
+        self.default_headers["Accept-Encoding"] = "gzip"
+    
+    def list(self, path):
+        """Retrieve a list of USS files based on a given pattern.
+
+        Returns
+        -------
+        json
+            A JSON with a list of dataset names matching the given pattern
+        """
```

Comment:
I know it may be a little redundant, and Timothy mentioned this offline not long ago...
Also, we don't have to do this as part of this PR.

Q: Should we make sure that the docs are complete and accurate?
  - For example, this is missing the parameters.

Reply:
I think we definitely need to standardize the docs, but maybe not in this PR

Reply:
Created #279 to track this